engine/pkg/externalresource/internal/file_manager.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	frameModel "github.com/pingcap/tiflow/engine/framework/model"
+	"github.com/pingcap/tiflow/engine/model"
+	resModel "github.com/pingcap/tiflow/engine/pkg/externalresource/model"
+	"github.com/pingcap/tiflow/engine/pkg/tenant"
+)
+
+// ResourceScope represents the environment in which the resource
+// has been or is to be created.
+type ResourceScope struct {
+	tenant.ProjectInfo
+
+	// Executor denotes the executor on which the resource is created.
+	Executor model.ExecutorID
+
+	// WorkerID denotes the worker that is creating or has created the resource.
+	// Note that it is NOT necessarily that consumes or depends on the resource.
+	WorkerID frameModel.WorkerID
+}
+
+// ResourceIdent provides information for the file manager to
+// uniquely determine where and how the resource is stored.
```

Comment:
The `WorkerID` should be unique in the system. Why do we need `ProjectInfo` and `ExecutorID` to identify a resource?

Reply:
This is a field reserved for multi-tenant resource isolation.